### ISSUE TITLE
Early return on missing deprecated functionality

### ISSUE THREAD

**Issue by GrimMaple** [BindBC/bindbc-opengl#48]
This code here:
```d
    with (GLSupport)
    static foreach(ver; [ gl12, gl13, gl14, gl15, gl20, gl21, gl30, gl31,
        gl32, gl33, gl40, gl41, gl42, gl43, gl44, gl45, gl46 ])
    {
        static if(ver <= glSupport)
        {
            // lib.loadGL30(contextVersion)
            if(mixin("lib.loadGL" ~ (cast(int)ver).stringof ~ "(contextVersion)"))
                loadedVersion = ver;
            else
                goto LoadARB;
        }
    }
```
Contains an early return that forces the `loadedVersion` to always be `gl11`. It happens on Intel UHD integrated video card because Intel graphics driver seems to be missing some of the deprecated functionality (I only tested 1.2, might have more missing stuff along the line). It seems to be incorrect to blindly assume 2.0 onwards stuff is missing if 1.2 functionality fails to load.

I'm not sure what to do in such a situation though. It is erroneous to assume that 2.0 can't be loaded because 1.2 deprecated failed to load. However, it can also lead to issues if errors are blindly ignored.

It feels to me the best options is to get some sort of `supports` function, which can be used to test if certain version of opengl successfully loaded

**Comment by ichordev** [comment]
I guess I’ll make it check each version and return the highest one instead of aborting at the first failure.

**Comment by GrimMaple** [comment]
> I guess I’ll make it check each version and return the highest one instead of aborting at the first failure.

It is up to you, but I see it as a bit of an ad-hoc, because you will get missing functions and random crashes trying to execute null pointers

**Comment by GrimMaple** [comment]
> > It is up to you, but I see it as a bit of an ad-hoc, because you will get missing functions and random crashes trying to execute null pointers
> 
> Oh, right. I guess it really needs to return a list of supported versions, but I don't want to break the API either...

The API doesn't really need to change; I think we only need a way to check if certain parts of OpenGL actually loaded

**Comment by ichordev** [comment]
> The API doesn't really need to change; I think we only need a way to check if certain parts of OpenGL actually loaded

Technically you can check if the function pointers are null manually... but in my opinion this is inadequate. I'm open to a PR or any further suggestions about how to resolve this.

**Comment by ichordev** [comment]
This should be fixed in v1.1.1.

**Comment by GrimMaple** [comment]
I have checked 1.1.1 on an Intel integrated card and this now works well.
